src/topology/sheaves/sheaf_condition/opens_le_cover.lean

```diff
@@ -324,6 +356,17 @@ lemma is_sheaf_iff_is_sheaf_pairwise_intersections :
 by rw [is_sheaf_iff_is_sheaf_opens_le_cover,
   is_sheaf_opens_le_cover_iff_is_sheaf_pairwise_intersections]
 
+lemma _root_.open_embedding.is_sheaf {f : Z ⟶ X} (hf : open_embedding f)
+  (h : F.is_sheaf) : is_sheaf (hf.is_open_map.functor.op ⋙ F) :=
+begin
+  rw is_sheaf_iff_is_sheaf_opens_le_cover at h ⊢,
+  intros ι W,
+  obtain ⟨hs⟩ := h (hf.is_open_map.functor.obj ∘ W),
+  refine ⟨is_limit.equiv_iso_limit _ $ hs.whisker_equivalence (hf.opens_le_cover_equivalence W).op⟩,
+  exact (cones.functoriality _ F).map_iso
+    ((cocone_equivalence_op_cone_op _).functor.map_iso $ hf.opens_le_cover_cocone_iso W).unop,
+end
```

Comment:
I prefer the following proof more, as the subproofs are of independent interest when we eventually have the API on geometric morphisms between toposes.
```suggestion
begin
  refine pullback_is_sheaf_of_cover_preserving ⟨_⟩ ⟨_⟩ ⟨_, h⟩,
  { intros ℱ U T x hx V₁ V₂ W i₁ i₂ i₃ i₄ hi₃ hi₄ e,
    let W' := (opens.map f).obj W,
    have : W.1 ⊆ set.range f := λ x h, by { obtain ⟨_,_,⟨⟩⟩ := i₁.le h, exact ⟨_, rfl⟩ },
    have : hf.is_open_map.functor.obj W' = W := opens.ext (set.image_preimage_eq_of_subset this),
    have hi₁ : W' ≤ V₁ := λ _ h, let h := (i₁.le h).some_spec in hf.inj h.2 ▸ h.1,
    have hi₂ : W' ≤ V₂ := λ _ h, let h := (i₂.le h).some_spec in hf.inj h.2 ▸ h.1,
    clear_value W',
    subst this,
    convert hx (hom_of_le hi₁) (hom_of_le hi₂) hi₃ hi₄ (subsingleton.elim _ _) },
  { rintros U S hU _ ⟨x, hx, rfl⟩,
    obtain ⟨V, i, hV, hxV⟩ := hU x hx,
    exact ⟨_, hf.is_open_map.functor.map i, ⟨_, i, 𝟙 _, hV, subsingleton.elim _ _⟩,
      set.mem_image_of_mem f hxV⟩ }
end
```

Reply:
This proof doesn't depend on opens_le_cover so could go into [this file](https://leanprover-community.github.io/mathlib_docs/topology/sheaves/sheaf_condition/sites.html) I think. And yes, we should extract `compatible_preserving` and `cover_preserving` as two lemmas, and maybe one of them hold under weaker conditions (`is_open_map`?). I'm not really familiar with these properties of functors between sites, so please feel free to commit directly or open another PR.

Reply:
I opened #17369.